src/main/java/org/zephyrsoft/sdb2/gui/SongCell.java

```diff
@@ -128,7 +131,11 @@ public void setImage(final String imageUrl, int degreesToRotateRight) {
 			this.image.setVisible(false);
 		} else {
 			try {
-				ImageIcon imageIcon = new ImageIcon(URI.create(imageUrl).toURL());
+				URI imageUri = URI.create(imageUrl);
+				if (imageUri.getScheme().equals("sdb")){
+					imageUri =  Paths.get(FileAndDirectoryLocations.getDBBlobDir(), imageUrl.replace("sdb://", "")).toUri();
```

Comment:
This is kind of duplicated by the code in PresenterWindow.java around line 239. Maybe a common method handling this replacement could be added? Maybe even more high-level in logic flow, like after loading the songs database?

BTW: It should also be checked if the newly referenced file exists - if not, this will fail when trying to display the image.